specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json

```diff
@@ -0,0 +1,2661 @@
+          "description": "Specifies the location of the encryption key in Key Vault."
+        }
+      }
+    },
+    "amlFilesystemHSMSettings": {
+      "description": "amlFilesystem HSM settings.",
+      "properties": {
+        "container": {
+          "description": "Container used for the amlFilesystem.",
+          "$ref": "#/definitions/URLString"
+        },
+        "containerKeyVaultKey": {
+          "x-ms-secret": true,
+          "$ref": "#/definitions/KeyVaultKeyReference",
+          "description": "Specifies the location of the encryption key in Key Vault."
+        },
+        "importPrefix" : {
+          "description": "Path/Prefix for blobs in above container.  Only blobs containing this prefix get pre-hydrated into the cluster namespace",
+          "type" : "string"
+        },
+        "exportPrefix" : {
+          "description": "Path/Prefix in above blob container where data gets exported when an archive task runs",
+          "type" : "string"
+        }
+      }
+    },
+    "amlFilesystemNamespaceHydrationStatus" : {
+      "description" : "Information about the amlFilesystem namespace hydration status",
+      "type" : "object",
+      "readOnly" : true,
+      "properties": {
+        "state" : {
+          "description" : "The state of the namespace hydration",
+          "type" : "string",
+          "enum" : [
+            "NotStarted",
+            "InProgress",
+            "Failed",
+            "Complete"
+          ],
+          "x-ms-enum": {
+            "name": "NamespaceHydrationStatusType",
+            "modelAsString": true
+          }
+        },
+        "progress" : {
+          "description" : "Progress information for the namespace hydration",
+          "type" : "object",
+          "properties" : {
+            "blobCount" : {
+              "description" : "The number of Blobs processed.",
+              "type": "integer"
+            },
+            "blobName" : {
+              "type" : "string",
+              "description" : "Name of the current blob"
+            },
+            "errorCount" : {
+              "type" : "integer",
+              "description" :"Number of failures"
+            }
+          }
+        }
+      }
+    },
+    "amlFilesystemArchive" : {
+      "description" : "Information about the amlFilesystem archive",
+      "type" : "object",
+      "readOnly": true,
+      "properties": {
+        "filesystemPath" : {
+          "readOnly": true,
+          "type" : "string",
+          "description" : "The path to the filesystem to archive."
+        },
+        "status" : {
+          "description" : "The status of the archive",
+          "type" : "object",
+          "readOnly" : true,
+          "properties": {
+            "state" : {
+              "readOnly": true,
+              "description" : "The state of the archive operation",
+              "type" : "string",
+              "enum" : [
+                "NotStarted",
```

Comment:
Instead of "NotStarted" and "Complete", how about these states:
"Idle", "InProgress"

To clarify "Idle", you already have "lastCompletionTime" to show that a job completed.

You can also add "lastStartedTime" so we can see a job duration.

Reply:
Updated.  See in next commit.

Reply:
I like the idea of canceling, but I need to figure out if robinhood allows you to gracefully cancel an archive request.  We can always do something like pkill -9.  I talked to BrianP, and since cancel is an action, it's probably a separate rest call (instead of a status).

Reply:
This suggests that the REST PUT call must support setting the state of the hydration to cancelled.

Reply:
Robinhood doesn't explicitly document a cancel operation, but it seems like it will be necessary.  As said above, we can always resort to a pkill of the process on the primary agent that runs the archive operation.  I suggest we add an archive cancel subop for this.

To represent this, I'm thinking we keep the "Idle" and "InProgress".  Call these "states".

Then add an additional "status" field, which has enum values of "Completed", "Canceled", "None".
(I'm suggestion "Completed" instead of "Success" because it could complete with errors.)

Here is how we use these:

Initial values for state/status:  Idle/None
While Running:  InProgress/None
Successful Completion:  Idle/Completed
Canceled: Idle/Canceled